sigal/plugins/compress_assets.py

```diff
@@ -0,0 +1,146 @@
+        compressed_filename = self.can_compress(filename)
+        if not compressed_filename:
+            return
+
+        self.do_compress(filename, compressed_filename)
+
+    def can_compress(self, filename):
+        if not os.path.splitext(filename)[1][1:] in self.settings['suffixes']:
+            return False
+
+        file_stats = None
+        compressed_stats = None
+        compressed_filename_result = self.compressed_filename(filename)
+        try:
+            file_stats = os.stat(filename)
+            compressed_stats = os.stat(compressed_filename_result)
+        except OSError: # FileNotFoundError is for Python3 only
+            pass
+
+        if file_stats and compressed_stats:
+            return compressed_filename_result if file_stats.st_mtime > compressed_stats.st_mtime else False
+        else:
+            return compressed_filename_result
+
+
+class GZipCompressor(BaseCompressor):
+    SUFFIX = 'gz'
+
+    def do_compress(self, filename, compressed_filename):
+        with open(filename, 'rb') as f_in, gzip.open(compressed_filename, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+
+class ZopfliCompressor(BaseCompressor):
+    SUFFIX = 'gz'
+
+    def do_compress(self, filename, compressed_filename):
+        import zopfli.gzip
+        with open(filename, 'rb') as f_in, open(compressed_filename, 'wb') as f_out:
+            f_out.write(zopfli.gzip.compress(f_in.read()))
+
+
+class BrotliCompressor(BaseCompressor):
+    SUFFIX = 'br'
+
+    def do_compress(self, filename, compressed_filename):
+        import brotli
+        with open(filename, 'rb') as f_in, open(compressed_filename, 'wb') as f_out:
+            f_out.write(brotli.compress(f_in.read(), mode=brotli.MODE_TEXT))
+
+
+def get_compressor(settings):
+    name = settings.get('method', '')
+    if name == 'gzip':
+        return GZipCompressor(settings)
+    elif name == 'zopfli':
+        try:
+            import zopfli.gzip
+            return ZopfliCompressor(settings)
+        except ImportError:
+            logging.warning('Zopfli not found, using standard gzip')
+            return GZipCompressor(settings)
+
+    elif name == 'brotli':
+        try:
+            import brotli
+            return BrotliCompressor(settings)
+        except ImportError:
+            logger.error('Unable to import brotli module')
+
+    else:
+        logger.error('No such compressor {}'.format(name))
+
+
+def compress_assets(assets_directory, compressor):
+    assets = []
+    for current_directory, _, filenames in os.walk(assets_directory):
+        for filename in filenames:
+            assets.append(os.path.join(current_directory, filename))
+
+    with progressbar(assets, label='Compressing theme assets') as progress_compress:
+        for filename in progress_compress:
+            compressor.compress_file(filename)
+
+
+def compress_gallery(gallery):
+    logging.info('Compressing assets for {}'.format(gallery.title))
```

Comment:
A minor remark, but good practice with logging is to use the "old" syntax, without explicit formatting, e.g. `logging.info('Compressing assets for %s', gallery.title)`. This way logging formats the message only if needed.